### PR TITLE
Build Docker image throught GitHub Actions

### PR DESCRIPTION
**Pull request by erseco**
Hello team,

I am excited to share that I have made a code contribution to our GitHub repository that will improve your workflow significantly. Specifically, I have added a new feature that allows us to build Docker images through GitHub Actions and push them to the GitHub Container Registry automatically.

I have thoroughly tested this code contribution and believe that it will be beneficial for your team. I am confident that it will streamline your workflow and save us time and effort in the long run. I hope that you find this contribution valuable as well.

Please review my pull request and let me know your thoughts. I am open to feedback and am eager to collaborate with you all to make our workflow even better.

Thank you for your time and consideration.

Best regards,
Ernesto

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2    
+
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
```
Comment (TheR1D) [inline .github/workflows/docker.yml:40]: This would need a token with write permission.

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+      - name: Build and push
+        uses: docker/build-push-action@v4
+        with:
+          platforms: linux/amd64
+          push: ${{ github.event_name != 'pull_request' }}
```
Comment (TheR1D) [inline .github/workflows/docker.yml:46]: Following lines can be updated according to suggestions above.

**Review by TheR1D — CHANGES_REQUESTED** [review]
Thank you for the PR, I left couple comments.
Also it would be good to update README.md with the new changes.

File: .github/workflows/docker.yml (hunk 1 above)
Comment (erseco) [reply .github/workflows/docker.yml:40]: The token mentioned in this context is automatically generated. You can find more information about this in the following link: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow. However, in order to utilize write permissions in your repository actions, you need to enable them in your action's settings. Please refer to the attached screenshot for guidance.
Captura de pantalla 2023-03-18 a las 11 14 19

**Comment by erseco** [comment]
Hi there,

I wanted to let you know that I have completed the changes you requested. I have thoroughly tested the code to ensure it meets the requirements and works as expected. Please review the changes and let me know if there is anything else you would like me to address. I'm looking forward to your feedback.

Thank you for your time and consideration.

Best regards.

**Comment by TheR1D** [comment]
@eric-glb FYI We are adding logic to build and publish containers when commit with tag is pushed to repository.

**Review by TheR1D — APPROVED** [review]
Thank you, good job.

**Comment by eric-glb** [comment]
Great, thanks Ernesto!

**Pull request merged**